torch/distributed/device_mesh.py

```diff
@@ -216,7 +237,11 @@ def __init__(
                 # already. The world pg is used for device mesh identity (rank) on each
                 # process (we need to know if the current global rank is in the mesh or not).
                 self._get_or_create_default_group()
-                self._init_process_groups()
+                # We need to temporarily bring _init_process_groups back, as we do not want to
+                # create new process group when slicing child mesh.
+                # TODO: Remove _init_process_groups after we change to reuse the subgroup when possible.
+                if _init_process_groups:
```

Comment:
Also, this original change (reverted) is introduced after 2.2 so it was not in any stable build yet. We will likely need some time to investigate the `_init_process_group` issue and find a solution.

Reply:
I think:
* It's fine to recreate pgs instead of adding the init_process_group again, as pg are lazy initialized (at least for now)
* we also immediately get rid of those created pgs with old pg info anyways, so I don't think it's a big problem

Reply:
This makes sense. Removing `_init_process_group` as a kwarg and for now we will go with recreate the pgs.